### DISCUSSION TITLE
Conditional Action based on execute query result in a form submission

### DISCUSSION THREAD

**Discussion by SantoshSrinivas79** [Budibase/budibase#4321] [Help] [answered]
Hello,

Is there a way I can do this?

i) Submit form using a REST API
ii) If the result is `{"success": "true", "next":"gohere"}` .. navigate to a new page using `next` param
iii) If the result is `{"success": "false"}` .. stay on the same page (ideally add an alert similar to `Error Occurred` or something)

**Comment by mjashanks** [comment, marked as answer]
You can use the result of the Execute Query in your `Navigate To`... e.g.

![image](https://user-images.githubusercontent.com/3524181/152612887-1ccafba7-eff9-4aa2-a7dc-4ee246c33bf3.png)


...you could show an error using `Update State`

**Reply by SantoshSrinivas79** [reply]
Thank you @mjashanks ... this seems pretty close! Will try out.
